pkg/pgmodel/ingestor/ingestor.go

```diff
@@ -25,35 +25,40 @@ type Cfg struct {
 
 // DBIngestor ingest the TimeSeries data into Timescale database.
 type DBIngestor struct {
-	db     model.Inserter
-	scache cache.SeriesCache
+	db           model.Inserter
+	scache       cache.SeriesCache
+	multiTenancy multi_tenancy_write.Authorizer
 }
 
 // NewPgxIngestorWithMetricCache returns a new Ingestor that uses connection pool and a metrics cache
 // for caching metric table names.
-func NewPgxIngestorWithMetricCache(conn pgxconn.PgxConn, cache cache.MetricCache, scache cache.SeriesCache, cfg *Cfg) (*DBIngestor, error) {
+func NewPgxIngestorWithMetricCache(conn pgxconn.PgxConn, cache cache.MetricCache, scache cache.SeriesCache, cfg *Cfg, mtAuthr multi_tenancy_write.Authorizer) (*DBIngestor, error) {
 	pi, err := newPgxInserter(conn, cache, scache, cfg)
 	if err != nil {
 		return nil, err
 	}
 
-	return &DBIngestor{db: pi, scache: scache}, nil
+	return &DBIngestor{
+		db:           pi,
+		scache:       scache,
+		multiTenancy: mtAuthr,
```

Comment:
```suggestion
		mtAuth: mtAuth,
```

Reply:
reason of such nomenclature:

`auth` => `authentication`
`authr` `authorizer`